doc/nrf/gs_programming.rst

```diff
@@ -122,3 +122,92 @@ After completing the :ref:`manual <build_environment_cli>` or :ref:`automatic <g
       The ``west flash`` command automatically resets the kit and starts the application.
 
 For more information on building and programming using the command line, see the Zephyr documentation on :ref:`zephyr:west-build-flash-debug`.
+
+.. _gs_programming_output:
+
+Output build files
+******************
+
+The building process produces each time an *image file*.
+
+.. output_build_files_info_start
+
+The image file, also referred to as *executable*, *program*, or *elf file*, consists of pieces of code and data that are identified by image-unique names recorded in a single symbol table.
+The symbol table exists as metadata in a :file:`.elf` or :file:`.exe` file and is not included when the image is converted to a HEX file for programming.
+Instead, a linker places the code and data at their addresses.
```

Comment:
Instead of what ? and when does it do it ?
How about rewriting the the lines into something like this:
```suggestion
As one of the last build steps, the linker will process all object files by locating code, data and symbols in sections in the final ELF file.
The linker replaces all symbol references to code and data with addresses.
A symbol table is created which maps addresses to symbol names, which is used by debuggers.

When an ELF file is converted into another format, such as HEX or binary, the symbol table is lost.
```
I left out some details in the above (the real world is a little more complex), but tried be accurate in as few words as possible.